tests/test_common.py

```diff
@@ -85,3 +85,20 @@ def test_to_audformat_dtype(dtype, expected):
 def test_to_pandas_dtype(dtype, expected):
     dtype = audformat.core.common.to_pandas_dtype(dtype)
     assert dtype == expected
+
+
+def test_items_order():
+
+    d = audformat.core.common.HeaderDict(sorted_iter=False)
+    d['b'] = 1
+    d['a'] = 0
+    assert dict(d.items()) == {'b': 1, 'a': 0}
+    assert list(d.keys()) == ['b', 'a']
+    assert list(d.values()) == [1, 0]
```

Comment:
I think we could make the output of `list()` consistent by overriding `__iter__()` and `__next__()`, but I guess you prefer that it does not match the output of `.keys()`.

Reply:
Exactly, I would prefer that it stays the way it is.
It also makes sense as the argument reads `sorted_iter`, so it makes sense if it affects only `iter()` and related methods.

Reply:
Well, one could argue that `__iter__()` is a related method :)

It basically means that e.g. the following is not affected by `sorted_iter=True`:

```
for scheme in schemes:
    ....
```

Reply:
But maybe it's not so important in the end. It's anyway kind of strange - we derive `HeaderDict` from `OrderedDict` so that the order is kept, but at the same time we have `sorted_iter=True` by default to return the items sorted alphabetically :) Then when we save a database we sort some parts like schemes, but others not like tables :)

Reply:
In the end the cleanest solution would be to switch to a sorted dict (if something like this exists) and remove `sorted_iter`. The consequence would be that everything (schemes, tables, etc) are always sorted.

Can you recall why we have schemes sorted and tables not?

Reply:
No, I cannot remember.

Reply:
Mhh ok. So maybe we should simply close this PR and continue #298 without it. We should then _think_ whether it makes sense to sort everything in the future so that there will be no difference with respect to the ordering when creating a database in memory or loading it from disk.

Reply:
I would also be fine with this pull request as it does not seem to affect the points I mentioned. But there is indeed no need for it in the release 0.15.2, so I would propose we first continue with #298.

Reply:
Ok, let's continue with #298 and put this one on hold as I think we need a clean solution where `list()`, `.items()`, 'pop()` etc. always has the same effect / result independent if a database is loaded from disk or not. Using a sorted dictionary seems to be the natural solution.

Reply:
Yes, sorted dictionary seems to make sense.

Reply:
I had a closer look and ...

1. We already sort tables alphabetically, e.g.:
  
  ```python
  db = audformat.testing.create_db(minimal=True)
  db['b'] = audformat.Table()
  db['a'] = audformat.Table()
  db
  ```
  ```
  name: unittest
  source: internal
  usage: unrestricted
  languages: [deu, eng]
  tables:
    a: {type: filewise}
    b: {type: filewise}
  ```
   
  The only items we currently do not sort are columns. If we want to sort those, too, we have to make sure to insert new columns in the underlying data frame at the proper position. And for backward compatibility we would need to sort the columns of a data frame after loading it. So I'm not sure if we should really change that.

2. There is no implementation of a sorted dictionary in the standard library. There is a [sortededcontainer](https://pypi.org/project/sortedcontainers/) package, but I don't think we want to introduce a new dependency for it. 

3. In https://github.com/audeering/audformat/issues/296 you argued that we should the order should be after assignment so that `popitem()` removes the last item that was added to the dictionary. But is there really a use-case for this? Or why would a user write a script where she e.g. adds a scheme and afterwards removes it again? The only use-case I see for removing a scheme is after loading an existing database and updating it. However, here we are already back to a sorted order anyway. Apart from that we should probably add a method `Database.drop_scheme()` that checks if the removed scheme is still used with one of the tables.

I now implemented also `__iter__()`, `__reversed__()` and `popitem()` and extended the tests. I also renamed `sort_iter` to `sort_by_key` to make it more obvious that they items in the dictionary are sorted by key.

Reply:
> The only items we currently do not sort are columns

I would also strongly vote against sorting column positions as a user might want to influence the order.

> But is there really a use-case for this?

The only usecase I see is an interactive session.

Reply:
> The only usecase I see is an interactive session.

And there the user can use `db.schemes.pop(scheme_id)` or better `db.drop_scheme(scheme_id)` if we decide to add such a method.

Reply:
Of cause there are other solutions, but at least you need to forbid `popitem()` or change it's behavior as the user cannot know that she is not supposed to use it.

Reply:
But you wrote already, that you implemented `popitem()` now, so I guess we are fine here anyway.